### PR TITLE
Update common classes

### PR DESCRIPTION
**Pull request by lkldev**
Update common classes to support getting the unfiltered lists

Update Module and ClassGroup to contain a list of students

Note: 
Storage needs to be updated to keep the list of students for class groups and modules

### CODE REVIEW

File: src/main/java/seedu/address/model/classgroup/ClassGroup.java [hunk 1]
```diff
@@ -16,21 +19,40 @@
...
      * @param taModule A valid module.
      */
     public ClassGroup(ClassGroupId classGroupId, ClassGroupType classGroupType, TaModule taModule) {
+        this(classGroupId, classGroupType, taModule, new UniqueStudentList());
+    }
+
+    /**
+     * Constructs a {@code ClassGroup}.
+     * Every field must be present and not null.
+     * Used to initialize a classgroup from storage file.
```
Comment (jxt00) [inline src/main/java/seedu/address/model/classgroup/ClassGroup.java:40]: Add a space between `classgroup`? Hahaha

Comment (lkldev) [reply src/main/java/seedu/address/model/classgroup/ClassGroup.java]: Resolved in latest commit

**Review by jxt00 — APPROVED** [review]
LGTM HAHAHA

**Review by wxliong — APPROVED** [review]
LGTM! :)

**Pull request merged**